dipy/io/tests/test_io_gradients.py

```diff
@@ -103,7 +104,29 @@ def test_read_bvals_bvecs():
         npt.assert_array_equal(ans, bvecs_3)
         npt.assert_array_equal(ans, bvecs_4)
 
-
-if __name__ == '__main__':
-    from numpy.testing import run_module_suite
-    run_module_suite()
+        bv_two_volume = 'bv_two_volume.txt'
+        with open(bv_two_volume, 'w') as f:
+            f.write("0 0 0\n0 0 0\n")
+        bval_two_volume = 'bval_two_volume.txt'
+        with open(bval_two_volume, 'w') as f:
+            f.write("0\n0\n")
+        bval_5, bvecs_5 = read_bvals_bvecs(bval_two_volume,
+                                           bv_two_volume)
+        npt.assert_array_equal(bvecs_5, np.zeros((2, 3)))
+        npt.assert_array_equal(bval_5, np.zeros(2))
+
+        bv_single_volume = 'test_single_volume.txt'
+        with open(bv_single_volume, 'w') as f:
+            f.write("0 0 0\n")
+        bval_single_volume = 'test_single_volume_2.txt'
+        with open(bval_single_volume, 'w') as f:
+            f.write("0\n")
+
+        with warnings.catch_warnings(record=True) as w:
+            bval_5, bvecs_5 = read_bvals_bvecs(bval_single_volume,
+                                               bv_single_volume)
+            npt.assert_array_equal(bvecs_5, np.zeros((1, 3)))
+            npt.assert_array_equal(bval_5, np.zeros(1))
+            assert_true(len(w) == 1)
+            assert_true(issubclass(w[0].category, UserWarning))
+            assert_true("Only 1 direction detected on" in str(w[0].message))
```

Comment:
Looks like this string needs to be updated to match the code. See https://travis-ci.com/github/dipy/dipy/jobs/401295692#L4905

Reply:
Indeed, thank you for the notice!